distribution/docs/src/main/jdocs/content/_developingComponents/developing-plugins-contents.adoc

```diff
@@ -0,0 +1,403 @@
+
+|Plugin Type
+|Plugin Interface
+|Invocation Order
+
+|<<_pre_authorization_plugins,Pre-Authorization>>
+|ddf.catalog.plugin.PreAuthorizationPlugin
+|Before any security rules are implemented.
+
+|<<_policy_plugins,Policy>>
+|ddf.catalog.plugin.PolicyPlugin
+|After pre-authorization plugins, but before other catalog plugins to establish the policy for requests/responses.
+
+|<<_access_plugins,Access>>
+|ddf.catalog.plugin.AccessPlugin
+|Directly after any policy plugins
+
+|<<_pre_ingest_plugins,Pre-Ingest>>
+|ddf.catalog.plugin.PreIngestPlugin
+|Before the Create/Update/Delete method is sent to the `CatalogProvider`.
+
+|<<_post_ingest_plugins,Post-Ingest>>
+|ddf.catalog.plugin.PostIngestPlugin
+|After the Create/Update/Delete method is sent to the Catalog Provider.
+
+|<<_post_process_plugins,Post-Process>>
+|org.codice.ddf.catalog.async.plugin.api.internal.PostProcessPlugin
+|After ingest for additional/asynchronous processing.
+
+|<<_pre_query_plugins,Pre-Query>>
+|ddf.catalog.plugin.PreQueryPlugin
+|Prior to the Query/Read method being sent to the Source.
+
+|<<_post_query_plugins,Post-Query>>
+|ddf.catalog.plugin.PostQueryPlugin
+|After results have been retrieved from the query but before they are posted to the Endpoint.
+
+|<<_pre_resource_plugins,Pre-Resource>>
+|ddf.catalog.plugin.PreResourcePlugin
+|Prior to a Resource being retrieved.
+
+|<<_post_resource_plugins,Post-Resource>>
+|ddf.catalog.plugin.PostResourcePlugin
+|After a Resource is retrieved, but before it is sent to the Endpoint.
+
+|<<_pre_create_storage_plugins,Pre-Create Storage>>
+|ddf.catalog.plugin.PreCreatePlugin
+|Before an item is created in the content repository.
+
+|<<_post_create_storage_plugins,Post-Create Storage>>
+|ddf.catalog.plugin.PostResourcePlugin
+|After an item is created in the content repository.
+
+|<<_pre_update_storage_plugins,Pre-Update Storage>>>
+|ddf.catalog.plugin.PostResourcePlugin
+|Before an item is updated in the content repository.
+
+|<<_post_update_storage_plugins,Post-Update Storage>>>
+|ddf.catalog.plugin.PostResourcePlugin
+|After an item is updated in the content repository.
+
+|<<_pre_subscription_plugins,Pre-Subscription>>
+|ddf.catalog.plugin.PreSubscriptionPlugin
+|Prior to a Subscription being created or updated.
+
+|<<_pre_delivery_plugins,Pre-Delivery>>
+|ddf.catalog.plugin.PreDeliveryPlugin
+|Prior to the delivery of a Metacard when an event is posted.
+
+|===
+
+==== Implement Plugins
+
+The procedure for implementing any of the plugins follows a similar format:
+
+. Create a new class that implements the specified plugin interface.
+
+. Implement the required methods.
+
+. Create an OSGi descriptor file to communicate with the OSGi registry.
+
+.. Register the plugin class as a service to OSGi registry.
+
+. Deploy to ${branding}.
+
+.Plugin Performance Concerns
+[NOTE]
+====
+Plugins should include a check to determine if requests are local or not. It is usually preferable to take no action on non-local requests.
```

Comment:
❓  Do we have some information we can link to in the docs to show how to check for local vs. remote requests?